src/main/java/com/productdock/library/inventory/application/service/InsertBookService.java

```diff
@@ -0,0 +1,19 @@
+package com.productdock.library.inventory.application.service;
+
+import com.productdock.library.inventory.application.port.in.InsertBookUseCase;
+import com.productdock.library.inventory.application.port.out.persistence.InventoryRecordsPersistenceOutPort;
+import com.productdock.library.inventory.domain.Inventory;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class InsertBookService implements InsertBookUseCase {
+
+    private InventoryRecordsPersistenceOutPort inventoryRecordRepository;
+
+    @Override
+    public void insertBook(Inventory bookInventory) {
+        inventoryRecordRepository.save(bookInventory);
```

Comment:
Can you please check what happens in case when inventory already has book with specific id? It looks like inventory will allow duplicated records. Can you make it rewrite book copies in case of a duplicate?

Reply:
It will rewrite book copies in case of duplicate and reserved and rented books count will be set to 0. Should I cover this case with a test?

Reply:
Moze test